src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Node/JsonNodeConverter.cs

```diff
@@ -6,14 +6,22 @@
 
 namespace System.Text.Json.Serialization.Converters
 {
-    internal class JsonNodeConverter : JsonConverter<object>
+    /// <summary>
+    /// Converter for JsonNode-derived types. The {T} value must be Object and not JsonNode
+    /// since we allow Object-declared members\variables to deserialize as {JsonNode}.
+    /// </summary>
+    internal sealed class JsonNodeConverter : JsonConverter<object?>
     {
+        private JsonArrayConverter? _arrayConverter;
+        private JsonObjectConverter? _objectConverter;
+        private JsonValueConverter? _valueConverter;
+
         public static JsonNodeConverter Instance { get; } = new JsonNodeConverter();
-        public JsonArrayConverter ArrayConverter { get; } = new JsonArrayConverter();
-        public JsonObjectConverter ObjectConverter { get; } = new JsonObjectConverter();
-        public JsonValueConverter ValueConverter { get; } = new JsonValueConverter();
+        public JsonArrayConverter ArrayConverter => _arrayConverter ??= new JsonArrayConverter();
+        public JsonObjectConverter ObjectConverter => _objectConverter ??= new JsonObjectConverter();
+        public JsonValueConverter ValueConverter => _valueConverter ??= new JsonValueConverter();
```

Comment:
Should these be static as well?

Reply:
Yes. Originally I didn't want the extra `if` statement since these are accessed for every JSON value so these were all created in the constructor. Then based on feedback I changed them to be lazy-loaded (but implies an extra `if` statement) but should have also made them static at that time.